### ISSUE TITLE
Policy file request failed when connecting to snowflake

### ISSUE THREAD

**Issue by DarkShoro** [Lekuruu/snowflake#42]
I attempted to setup snowflake for my Card Jitsu snow to work, I followed all the steps to get it working, and when I attempt to load the game, I get the "already logged in" error with the message "WARNING: Connection lost: Connection was closed cleanly." popping in Snowflake's console.

**Comment by ERmilburn02** [comment]
Have you tried enabling debug logging in the env file to see what exactly what happened?

**Comment by DarkShoro** [comment]
Yes!
```
[2024-06-09 13:40:02,423] - <Snowflake> DEBUG: -> "XX.XX.XX.XX:54151"
[2024-06-09 13:40:02,424] - <XX.XX.XX.XX> DEBUG: -> "<policy-file-request/>"
[2024-06-09 13:40:02,539] - <Snowflake> DEBUG: -> "XX.XX.XX.XX:54152"
[2024-06-09 13:40:02,539] - <XX.XX.XX.XX> WARNING: Connection lost: Connection was closed cleanly.
```

But that doesn't help me at all :(

**Comment by ERmilburn02** [comment]
Looks like the server isn't responding with the policy file. @lekuruu Any ideas what could be causing this?

**Comment by DarkShoro** [comment]
Do you have any troubleshootings steps I could try in the meantime?

**Comment by ERmilburn02** [comment]
@DarkShoro I'm not too sure, given that the client connects and requests the policy file, I don't know what could be wrong. Levi knows a lot more about the way it connects so I'm afraid I'll have to wait for him. In the meantime, you can just try some basic stuff like reinstalling dependencies and making sure your config is set correctly.

**Comment by DarkShoro** [comment]
Maybe I need to set "ENABLE_POLICY_SERVER" to true?

I've seen it set to false.

**Comment by DarkShoro** [comment]
Oh my god that was it!

Configuring

```
ENABLE_POLICY_SERVER=True
POLICY_DOMAIN="*"
POLICY_PORT="*"
```
and restarting the server did help and now the game load fine!

**Comment by Lekuruu** [comment]
I've left the Flash Policy Server there as a backup for this exact reason. I am glad it worked!

**Comment by ERmilburn02** [comment]
@Lekuruu but if it's a backup, then what was the main way meant to be? And why didn't it work this time?

**Comment by Lekuruu** [comment]
I think this should answer your question: https://superuser.com/questions/617385/any-need-to-open-adobe-socket-policy-server-port-843

tl;dr, the flash client always does a policy request as a fallback on the target socket server, if the policy server on port 843 is not available.
You can check how snowflake handles this [here](https://github.com/Lekuruu/snowflake/blob/main/app/protocols/metaplace/protocol.py#L47-L54). Houdini also does the same thing [here](https://github.com/solero/houdini/blob/master/houdini/spheniscidae.py#L115-L116). There have been a few people who had this issue while setting up snowflake, so I've decided to write a policy server which you can enable in the config.

**Comment by Lekuruu** [comment]
I have added some troubleshooting steps to the setup guide. That should hopefully do it.

**Comment by ERmilburn02** [comment]
@Lekuruu let's hope so
